nav2-usability/scenario-code/dynamic-linking/dynamic_linking_vrouter.dart

```diff
@@ -0,0 +1,159 @@
+// Copyright 2021, the Flutter project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// Dynamic linking example
+/// Done using VRouter
+
+import 'dart:math';
+
+import 'package:flutter/material.dart';
+import 'package:vrouter/vrouter.dart';
+
+void main() {
+  runApp(WishlistApp());
+}
+
+class Wishlist {
+  final String id;
+
+  Wishlist(this.id);
+}
+
+class AppState extends ChangeNotifier {
+  final List<Wishlist> wishlists = <Wishlist>[];
+
+  void addWishlist(Wishlist wishlist) {
+    wishlists.add(wishlist);
+    notifyListeners();
+  }
+}
+
+class WishlistApp extends StatefulWidget {
+  @override
+  _WishlistAppState createState() => _WishlistAppState();
+}
+
+class _WishlistAppState extends State<WishlistApp> {
+  final AppState _appState = AppState();
+  final vRouterKey;
+
+  _WishlistAppState() : vRouterKey = GlobalKey<VRouterState>() {
+    _appState.addListener(() {
+      setState(() {});
+    });
+  }
+
+  @override
+  void dispose() {
+    _appState.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return VRouter(
+      key: vRouterKey,
+      routes: [
+        VWidget(
+          path: '/',
+          widget: WishlistListScreen(wishlists: _appState.wishlists, onCreate: onCreate),
+          stackedRoutes: [
+            VGuard(
+              beforeEnter: (vRedirector) async => checkIfValid(vRedirector.newVRouterData!.pathParameters['id']!),
+              beforeUpdate: (vRedirector) async => checkIfValid(vRedirector.newVRouterData!.pathParameters['id']!),
+              stackedRoutes: [
+                VWidget(
+                  path: r'wishlist/:id(\d+)',
+                  widget: Builder(
+                    builder: (context) => WishlistScreen(
+                        wishlist: _appState.wishlists.firstWhere(
+                                (element) => element.id == context.vRouter.pathParameters['id'])),
+                  ),
+                ),
+              ],
+            ),
+          ],
+        ),
+      ],
+    );
+  }
+
+  void onCreate(String value) {
+    final wishlist = Wishlist(value);
+    _appState.addWishlist(wishlist);
+    vRouterKey.currentState!.push('/wishlist/$value');
+  }
+
+  void checkIfValid(String value) {
```

Comment:
nit: maybe call createIfNotExist

Reply:
Much better, done in 5297811de6a38beacc8b48df0b49d7c8973716b6.